projects/charger/inc/generic_can_network.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+// Module to abstract the network CAN variant such that it shares an interface with UART CAN.
+// Requires network layer CAN to be running.
+
+#include "generic_can.h"
+
+#define NUM_GENERIC_CAN_NETWORK_RX_HANDLERS 5
+
+typedef struct GenericCanNetwork {
+  GenericCan base;
+  GenericCanRxStorage rx_storage[NUM_GENERIC_CAN_NETWORK_RX_HANDLERS];
+} GenericCanNetwork;
+
+// Initializes the generic CAN for the hardware CAN.
+StatusCode generic_can_network_init(GenericCanNetwork* out);
```

Comment:
Hmm, this seems a little out of place. CAN HW and CAN-UART make sense, since they're very similar levels of abstraction, but this doesn't seem to expose the ACK handling and I'm a little confused about how IDs are handled here vs. the actual CAN network layer. Here, it seems like filtering an ID would filter the raw ID value, but in the network layer it will filter the message ID. It seems a little inconsistent.

Is there a benefit to supporting this layer?

Reply:
I'll clean up the ID management to be more consistent. The main benefit is it allows the development of modules like `can_interval` (see PR #142) which are completely oblivious to the underlying CAN implementation.

Reply:
Other than drive commands (which has its own reliability-oriented interval output) and periodic telemetry, I believe our other interval-based CAN messages on the network layer use ACK handling to keep track of who's responded (ex. heartbeats).

Reply:
A lot of data messages on for telemetry will be periodic (Chaos and Charger each have at least one) I expect the IMU to have this as well. I think it is worth including the network layer for that abstraction. Using `GenericCan*` and `CanInterval` is optional so if you just send network messages it is easy to roll your own. This is really just an optional helper I'd like to add for my own usage in various boards (others may or may not find it useful).